src/cairo/bootloader.md

```diff
@@ -1 +1,199 @@
+      ecdsa=2,
+      bitwise=5,
+      ec_op=7,
+      keccak=16,
+      poseidon=6,
+    ```
+
+  - So for example, since the pedersen builtin requires 3 cells (2 for inputs, 1 for output), the bootloader checks that the difference between the pedersen builtin pointer before and after the task program run is positive (this is where the range check comes in) and that it is divisible by 3.
+
+#### How are task program bytecode loaded into memory?
+
+The bootloader creates a separate segment in the memory for storing each task program bytecode
+
+#### How are task outputs written to memory?
+
+The bootloader writes the task outputs sequentially (i.e. in the order that the task programs are run) to the output builtin segment. Below is an example of what this looks like:
+
+<div style="text-align: center;">
+    <img src="simple_bootloader_output_memory_layout.png" alt="Simple bootloader output memory layout" width="100%">
+</div>
+
+#### How do task programs use builtins other than the output builtin?
+
+- assuming that the selected layout supports the builtins that the task program uses, the bootloader will pass the necessary builtin pointers to the task program's main function, and the task program will use builtin cells as needed and return the pointers to the next unused builtin cells.
+- at the end of the task program run, the bootloader will validate that the builtin pointers have been correctly advanced, as explained in the [What builtins does the bootloader use?](#what-builtins-does-the-bootloader-use) section.
+
+#### How are task program hints handled?
+
+- As mentioned in the [A short primer on how the CairoVM works](#a-short-primer-on-how-cairovm-works) section, hints are specified to run before a certain bytecode is executed. In CairoVM, that refers the the “pc” value, or the program counter, and this value is normally a value in Segment 0. So a typical hint should run before, say when pc=(Segment 0, Index 15).
+- However, task program bytecode are loaded into a separate segment (e.g. Segment 11), so the pc values for the hints of the task program need to updated accordingly.
+
+## How the "advanced" bootloader works
+
+The "advanced" bootloader builds upon the "simple" bootloader by allowing as input not just Cairo programs but also a structured set of inputs that include both Cairo programs and simple bootloader proofs. In this section, we will go over how this works in multiple steps.
+
+(The "advanced" bootloader refers to the `bootloader.cairo` file ([link](https://github.com/starkware-libs/cairo-lang/blob/v0.13.1/src/starkware/cairo/bootloaders/bootloader/bootloader.cairo)) in the Cairo repo.)
+
+### Step 1: A Cairo verifier program
+
+First, we need to understand what a Cairo verifier program is.
+
+A Cairo verifier is a Cairo program that verifies a Cairo proof, and in this case, it will only verify a proof generated using the "simple" bootloader. Below is an example of what running a Cairo verifier program looks like.
+
```

Comment:
How do I keep recursing? 

If the Cairo verifier can only verify simple bootloader proofs and Cairo verifier proofs can only be bootloaded with the advanced bootloader, then I can't recurse on the STARK proof of the Cairo verifier?

Reply:
Cairo verifier proofs can be bootloaded with the simple bootloader as well 🙂